src/MoBi.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -0,0 +1,71 @@
+﻿using MoBi.Core.Serialization.Xml.Services;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Snapshots.Mappers;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Utility.Extensions;
+using ModelProject = MoBi.Core.Domain.Model.MoBiProject;
+using SnapshotProject = MoBi.Core.Snapshots.Project;
+
+namespace MoBi.Core.Snapshots.Mappers;
+
+public class ProjectMapper : SnapshotMapperBase<ModelProject, SnapshotProject, ProjectContext>
+{
+   private readonly IXmlSerializationService _xmlSerializationService;
+   private readonly ICreationMetaDataFactory _creationMetaDataFactory;
+
+   public ProjectMapper(IXmlSerializationService xmlSerializationService, ICreationMetaDataFactory creationMetaDataFactory)
+   {
+      _xmlSerializationService = xmlSerializationService;
+      _creationMetaDataFactory = creationMetaDataFactory;
+   }
+      
+   public override Task<ModelProject> MapToModel(SnapshotProject snapshot, ProjectContext context)
+   {
+      var project = new ModelProject
+      {
+         Name = snapshot.Name,
+         Description = snapshot.Description,
+         Creation = _creationMetaDataFactory.Create()
+      };
+         
+      snapshot.ExtensionModules?.Each(x => project.AddModule((deserializeFromBase64PKML<Module>(x, project))));
+      
+      snapshot.ExpressionProfileBuildingBlocks?.Each(x => project.AddExpressionProfileBuildingBlock(deserializeFromBase64PKML<ExpressionProfileBuildingBlock>(x, project)));
+      
+      snapshot.IndividualBuildingBlocks?.Each(x => project.AddIndividualBuildingBlock(deserializeFromBase64PKML<IndividualBuildingBlock>(x, project)));
+
+      return Task.FromResult(project);
```

Comment:
and where are PK-Sim modules created?

Reply:
But at the end it should also become part of `MapToModel`? OR it will be a fully separate step?

Reply:
Yeah, It should be done right here. Along with simulations, charts everything required to reproduce the project

Reply:
It will become very interesting for individuals and expression profiles... Looking forward 😄